src/version-util.ts

```diff
@@ -52,17 +43,11 @@ export async function getLatestGcloudSDKVersion(): Promise<string> {
  * should be called without parameters.
  *
  */
-async function getGcloudVersion(url: string): Promise<string> {
-  const http = new httpm.HttpClient(userAgentString);
-  const res = await http.get(url);
-  if (res.message.statusCode && res.message.statusCode != 200) {
-    throw new Error(`Failed to retrieve gcloud SDK version, statusCode: ${res.message.statusCode}`);
-  }
-
-  const body = JSON.parse(await res.readBody());
-  const version = body.version;
-  if (!version) {
-    throw new Error(`Failed to retrieve gcloud SDK version, invalid response body: ${body}`);
+export async function getGcloudVersion(url: string): Promise<string> {
+  const http = new HttpClient(userAgentString, undefined, { allowRetries: true, maxRetries: 3 });
+  const res = await http.getJson<{ version: string }>(url);
```

Comment:
It doesn't look like the library actually checks 400s or 500s. I think we need to get the body first, then check the response, then print the body on error. Otherwise we risk swallowing the message.

Reply:
The getJson does [check](https://github.com/actions/http-client/blob/5ea1c5292929dba26848ea46f3b4c2e1ea1e30a2/index.ts#L746) here but since we want op I will switch to the other approach.

Reply:
Oh that's a weird API. The README specifically states that errors are not checked. And it looks like that's true for non-JSON responses.